queries/aws_queries/ec2_crud_activity_by_role_query.yml

```
@@ -0,0 +1,18 @@
+AnalysisType: scheduled_query
+QueryName: Query.EC2.CRUD.Activity.Role
+Enabled: false
+Description: >
+  This query searches for CRUD activity in EC2 by role arn. Activites from a role outside typical deployment processes may warrant investigation. 
+SnowflakeQuery: >
+  SELECT count(*),recipientAccountId, userIdentity:arn, eventName, eventSource
+  FROM panther_logs.public.aws_cloudtrail
+  WHERE 
+    eventSource = 'ec2.amazonaws.com' 
+    AND eventName LIKE '%Image%'
+    AND eventName NOT LIKE '%Describe%'
+    AND p_occurs_since('7 day')
```

If this query runs daily, why would we query 7 days' worth of data? 
```suggestion
    AND p_occurs_since('1 day')
```

---

It would be really good to also check that we don't have this issue with other queries.

---

> If this query runs daily, why would we query 7 days' worth of data?

We've been talking about using this query to kick off the investigative process. We figured to start the query with 7 days worth of data to gather a big enough population of events to look for population data. 1 day might be enough depending on how frequently folks are building/modifying AMIs.

---

My idea here was that this query is more of an investigative tool to catch unusual useragents/roles being used so having some historic data of what might be 'normal' included in the query to find the anomalies would be helpful. Maybe 7 days is too much but maybe 3 days?

---

If the intention is then to just write a query - then it shouldn't be scheduled, otherwise it'll be running in the background every day with nobody looking at the results